gcc/rust/typecheck/rust-hir-type-check-implitem.cc

```diff
@@ -389,12 +389,14 @@ TypeCheckImplItemWithTrait::visit (HIR::ConstantItem &constant)
 				       TraitItemReference::TraitItemType::CONST,
 				       &raw_trait_item);
 
-  // unknown trait item
+  // unknown trait item - https://doc.rust-lang.org/error_codes/E0323.html
   if (!found || raw_trait_item->is_error ())
     {
       rich_location r (line_table, constant.get_locus ());
       r.add_range (trait_reference.get_locus ());
-      rust_error_at (r, "constant %<%s%> is not a member of trait %<%s%>",
+      rust_error_at (r, ErrorCode ("E0323"),
+		     "item %<%s%> is an associated const, which does not match "
```

Comment:
- Refactored Error message, similiar to rustc.
- With a little difference, 
- The diff of rustc and gccrs error is:

```diff
# gccrs
+ item ‘N’ is an associated const, which does not match its trait ‘Foo’  

- item `N` is an associated const, which doesn't match its trait `Foo`
```

Is there a way to add single apostrophe, in the error message:

```bash
../../gccrs/gcc/rust/typecheck/rust-hir-type-check-implitem.cc:398:68: warning: bare apostrophe ‘'’ in format [-Wformat-diag]
  398 |                      "item %<%s%> is an associated const, which doe's not match "
      |                                                                    ^
../../gccrs/gcc/rust/typecheck/rust-hir-type-check-implitem.cc:398:22: note: if avoiding the apostrophe is not feasible, enclose it in a pair of ‘%<’ and ‘%>’ directives instead
  398 |                      "item %<%s%> is an associated const, which doe's not match "
      |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  399 |                      "its trait %<%s%>",
      |                      ~~~~~~~~~~~~~~~~~~

```

Reply:
the grammar you used is correct, no need to add a single apostrophe?

Reply:
@CohenArthur, Yeah there is no need to add a single apostrophe. But it would be better to add, then the whole message will be exactly similiar to rustc..